docs/source/reconciliation.rst

```diff
@@ -0,0 +1,154 @@
+2. Add the record's old preferred prefix to the record's prefix synonyms
+
+This means applying the CURIE prefix remapping ``{"a": "c"}`` results in the following
+
+.. code-block:: json
+
+    [
+        {"prefix": "c", "uri_prefix": "https://example.org/a/", "prefix_synonyms": ["a", "a1"]},
+        {"prefix": "b", "uri_prefix": "https://example.org/b/"}
+    ]
+
+Similarly, if the old prefix appears in a record in the extended prefix map as a prefix synonym, do the same.
+This means applying the CURIE prefix remapping ``{"a1": "c"}`` results in the following
+
+.. code-block:: json
+
+    [
+        {"prefix": "c", "uri_prefix": "https://example.org/a/", "prefix_synonyms": ["a", "a1"]},
+        {"prefix": "b", "uri_prefix": "https://example.org/b/"}
+    ]
+
+3. New prefix doesn't exist, old prefix doesn't exist
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+If neither the old prefix nor new prefix appear in the extended prefix maps, one of two things can happen:
+
+1. Do nothing (lenient)
+2. Raise an exception (strict)
+
+URI Prefix Remapping
+----------------------
+URI prefix remapping is configured by a mapping from existing URI prefixes to new URI prefixes.
+The rules work exactly the same as with CURIE prefix remapping, but for the :data:`curies.Record.uri_prefix` and
+:data:`curies.Record.uri_prefix_synonyms` fields.
+
+Prefix Rewiring
+---------------
+Prefix rewiring is configured by a mapping from existing CURIE prefixes to new URI prefixes.
+The following rules are applied for each pair of CURIE prefix/URI prefix:
+
+1. CURIE prefix doesn't exist
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+If the CURIE prefix doesn't exist in the extended prefix map, then the pair is simply appended.
+This means applying the rewiring ``{"c": "https://example.org/c"}`` results in the following
+
+.. code-block:: json
+
+    [
+        {"prefix": "a", "uri_prefix": "https://example.org/a/", "prefix_synonyms": ["a1"]},
+        {"prefix": "b", "uri_prefix": "https://example.org/b/"},
+        {"prefix": "c", "uri_prefix": "https://example.org/c/"}
+    ]
+
+2. CURIE prefix exists, URI prefix doesn't exist
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+If the CURIE prefix appears as either the preferred prefix or a prefix synonym, do the following
+
+1. Replace the record's preferred URI prefix with the new URI prefix
+2. Add the record's old preferred URI prefix to the record's URI prefix synonyms
+
+This means applying the rewiring ``{"b": "https://example.org/b_new/"}`` results in the following
+
+.. code-block:: json
+
+    [
+        {"prefix": "a", "uri_prefix": "https://example.org/a/", "prefix_synonyms": ["a1"]},
+        {"prefix": "b", "uri_prefix": "https://example.org/b_new/", "uri_prefix_synonyms": ["https://example.org/b/"]}
+    ]
+
+3. URI prefix exists
+~~~~~~~~~~~~~~~~~~~~
+If the URI prefix appears as either a preferred URI prefix or as a URI prefix synonym in
+any record in the extended prefix map, do one of the following:
```

Comment:
You may also want to distininguish the case where the URI prefix exists as a uri_prefix synonym corresponding to the same CURIE prefix, which would be a "safe" operation.

Reply:
I know you call this "remapping" but maybe at least mention in docs?